### PR TITLE
Milestone 09 : Add Devise

### PR DESCRIPTION
**Pull request by FaranoshAmini**
## In this pull request, we have implemented the following changes:

- [x]  Installed and set up devise to:
- [x]  register a new user
- [x]  Log in user with a combination of email and password.
- [x]  Store hashed password in the database.
- [x]  Ask for confirmation of email.
- [x]  Allow user to reset password.
- [x]  Modified views for the registration and login

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (DeliceLydia) [inline app/views/devise/sessions/new.html.erb:26]: - [x] I am able to log in but for some reason, I am getting an empty page
check the reason

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @FaranoshAmini, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Added devise
- No linters errors


## Required Changes ♻️
_Check the comments under the review._


### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (AmineDerbal) [reply app/views/devise/sessions/new.html.erb:26]: @DeliceLydia 

- Can you give a screenshot of the view in the browser 
- When I log in I am redirected to /users index page

Comment (FaranoshAmini) [reply app/views/devise/sessions/new.html.erb:26]: @DeliceLydia, When I sign in I am redirected to the /users index page
image

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @FaranoshAmini , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  App working as expected 
:heavy_check_mark: Excellent communication with reviewers and adding screenshot to PR comment

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**